src/components/ui/swapContainer.js

```diff
@@ -178,6 +178,7 @@ const SwapContainer = ({
   const handleOnInputChange = event => {
     let { value } = event.target;
     value = Number(value);
+    value = Number(value.toString().match(/^-?\d+(?:\.\d{0,2})?/)[0]);
```

Comment:
are the numbers being formatted based on user intl ? because this can be a problem if users uses , instead of . to separate decimals and so on.

perhaps there is a better way to do this without assuming that regexp is gonna work

Reply:
Searching an alternative...

Reply:
I was thinking you could use parseFloat(xx).toFixed(2) but doing the parseFloat you are gonna hit the same rounding issue

Reply:
Yep. Using .toFixed() will lead us to the same issue. I played around with it

Reply:
@kesar I think the issue is solved